bumble/device.py

```diff
@@ -433,37 +443,344 @@ def is_directed(self):
+AdvertisingChannelMap = HCI_LE_Set_Extended_Advertising_Parameters_Command.ChannelMap
 
 
 # -----------------------------------------------------------------------------
 @dataclass
-class ExtendedAdvertiser(CompositeEventEmitter):
+class AdvertisingParameters:
+    # pylint: disable=line-too-long
+    advertising_event_properties: AdvertisingEventProperties = field(
+        default_factory=AdvertisingEventProperties
+    )
+    primary_advertising_interval_min: int = DEVICE_DEFAULT_ADVERTISING_INTERVAL
+    primary_advertising_interval_max: int = DEVICE_DEFAULT_ADVERTISING_INTERVAL
+    primary_advertising_channel_map: HCI_LE_Set_Extended_Advertising_Parameters_Command.ChannelMap = (
+        AdvertisingChannelMap.CHANNEL_37
+        | AdvertisingChannelMap.CHANNEL_38
+        | AdvertisingChannelMap.CHANNEL_39
+    )
+    own_address_type: OwnAddressType = OwnAddressType.RANDOM
+    peer_address: Address = Address.ANY
+    advertising_filter_policy: int = 0
+    advertising_tx_power: int = DEVICE_DEFAULT_ADVERTISING_TX_POWER
+    primary_advertising_phy: Phy = Phy.LE_1M
+    secondary_advertising_max_skip: int = 0
+    secondary_advertising_phy: Phy = Phy.LE_1M
+    advertising_sid: int = 0
+    enable_scan_request_notifications: bool = False
+    primary_advertising_phy_options: int = 0
+    secondary_advertising_phy_options: int = 0
+
+
+# -----------------------------------------------------------------------------
+@dataclass
+class PeriodicAdvertisingParameters:
+    # TODO implement this class
+    pass
+
+
+# -----------------------------------------------------------------------------
+@dataclass
+class AdvertisingSet(EventEmitter):
     device: Device
-    handle: int
-    advertising_properties: HCI_LE_Set_Extended_Advertising_Parameters_Command.AdvertisingProperties
-    own_address_type: OwnAddressType
+    advertising_handle: int
     auto_restart: bool
-    advertising_data: Optional[bytes]
-    scan_response_data: Optional[bytes]
+    random_address: Optional[Address]
+    advertising_parameters: AdvertisingParameters
+    advertising_data: bytes
+    scan_response_data: bytes
+    periodic_advertising_parameters: Optional[PeriodicAdvertisingParameters]
+    periodic_advertising_data: bytes
+    selected_tx_power: int = 0
+    enabled: bool = False
 
     def __post_init__(self) -> None:
         super().__init__()
 
+    async def set_advertising_parameters(
+        self, advertising_parameters: AdvertisingParameters
+    ) -> None:
+        # Sanity check
+        if (
+            not advertising_parameters.advertising_event_properties.is_legacy
+            and advertising_parameters.advertising_event_properties.is_connectable
+            and advertising_parameters.advertising_event_properties.is_scannable
+        ):
+            raise ValueError(
+                "non-legacy extended advertising event properties may not be both "
+                "connectable and scannable"
+            )
```

Comment:
I prefer to raise warning instead of error here? Also, before issuing a configuration command, we may also raise sanity check warnings.

BTW, the word "sanity" is not considered inclusive in Google. Might be better to call it like compliance?

Reply:
Done